servlet/src/main/java/com/linecorp/armeria/server/servlet/DefaultServletService.java

```diff
@@ -0,0 +1,89 @@
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.servlet;
+
+import static java.util.Objects.requireNonNull;
+
+import javax.servlet.http.HttpServlet;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpResponseWriter;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.ResponseHeaders;
+import com.linecorp.armeria.server.HttpService;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+/**
+ * An {@link HttpService} which handles {@link HttpRequest} and forward to Servlet APIs,
+ * and write {@link HttpResponse} to client.
+ */
+public class DefaultServletService implements HttpService {
+    private static final Logger logger = LoggerFactory.getLogger(DefaultServletService.class);
+    private final ServletContextImpl servletContext;
+
+    /**
+     * A class which helps a {@link DefaultServletService} have a {@link HttpServlet}.
+     */
+    public DefaultServletService(ServletContextImpl servletContext) {
+        requireNonNull(servletContext, "servletContext");
+        this.servletContext = servletContext;
+    }
+
+    /**
+     * handles {@link HttpRequest} and forward to Servlet APIs.
+     */
+    @Override
+    public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) throws Exception {
+        final HttpResponseWriter res = HttpResponse.streaming();
+        req.aggregate().handle((aReq, cause) -> {
+            if (cause != null) {
+                logger.warn("{} Failed to aggregate a request:", ctx, cause);
+                if (res.tryWrite(ResponseHeaders.of(HttpStatus.INTERNAL_SERVER_ERROR))) {
+                    res.close();
+                }
+                return null;
+            }
+            process(res, ServletHttpExchange.of(servletContext, ctx, aReq));
+            return null;
+        });
+        return res;
+    }
+
+    private void process(HttpResponseWriter res, ServletHttpExchange servletHttpExchange) {
+        final ServletHttpRequest httpServletRequest = servletHttpExchange.getRequest();
+        final ServletHttpResponse httpServletResponse = servletHttpExchange.getResponse();
+        try {
+            final ServletRequestDispatcher dispatcher =
+                    servletHttpExchange.getServletContext()
+                                       .getRequestDispatcher(httpServletRequest.getRequestURI());
```

Comment:
If we add the `ServletService` one by one, https://github.com/line/armeria/pull/2686/files#diff-6b0e2525eb80747c141e9fb8daa2c3adR56, we don't need this dispatcher
because Armeria server will handle the routing.

Reply:
Yes, we don't need routing here. But dispatcher help us insert filter class (interceptor). I think we should keep dispatcher here.

Reply:
@dominhhien Thanks for the explanation. Honestly, I'm not sure if we need this or not. If it's just only for the filter, we can implement our own logic for that. so that we can remove `UrlMapper` and `MimeMappings` classes that Armeria server already handles.
Is there any other usage for the `RequestDiscpatcher` except for the filter?

Reply:
Also, is there any chance that a user needs to use `RequestDispatcher`?